pkg/controller/v1beta1/inferenceservice/controller_test.go

```diff
@@ -474,6 +474,394 @@ var _ = Describe("v1beta1 inference service controller", func() {
+					Status: "True",
+				},
+			}
+			copiedKsvc.Status.LatestCreatedRevisionName = "revision-v1"
+			copiedKsvc.Status.LatestReadyRevisionName = "revision-v1"
+			Expect(k8sClient.Status().Update(context.TODO(), copiedKsvc)).NotTo(HaveOccurred())
+
+			// Check that the virtual service is present
+			virtualService := &istioclientv1beta1.VirtualService{}
+			Eventually(func() bool {
+				err := k8sClient.Get(context.TODO(), serviceKey, virtualService)
+				return err == nil
+			}, timeout).
+				Should(BeTrue())
+
+			// Check that the service is present
+			service := &corev1.Service{}
+			Eventually(func() bool {
+				err := k8sClient.Get(context.TODO(), serviceKey, service)
+				return err == nil
+			}, timeout).
+				Should(BeTrue())
+
+			// Check that the stopped condition is false
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, serviceKey, updatedIsvc)
+				if err == nil {
+					stopped_cond := updatedIsvc.Status.GetCondition(v1beta1.Stopped)
+					if stopped_cond != nil && stopped_cond.Status == corev1.ConditionFalse {
+						return true
+					}
+				}
+				return false
+			}, timeout, interval).Should(BeTrue(), "The stopped condition should be set to false")
+		})
+
+		It("Should delete the knative service when the annotation is set to true", func() {
+			ctx, cancel := context.WithCancel(context.Background())
+			DeferCleanup(cancel)
+
+			// Config map
+			configMap := createInferenceServiceConfigMap()
+			Expect(k8sClient.Create(context.TODO(), configMap)).NotTo(HaveOccurred())
+			defer k8sClient.Delete(ctx, configMap)
+
+			// Serving runtime
+			serviceName := "stop-true-isvc"
+			serviceNamespace := "default"
+			expectedRequest := reconcile.Request{NamespacedName: types.NamespacedName{Name: serviceName, Namespace: serviceNamespace}}
+			serviceKey := expectedRequest.NamespacedName
+			storageUri := "s3://test/mnist/export"
+
+			servingRuntime := createServingRuntime(serviceKey.Namespace, "tf-serving")
+			Expect(k8sClient.Create(context.TODO(), servingRuntime)).NotTo(HaveOccurred())
+			defer k8sClient.Delete(ctx, servingRuntime)
+
+			// Define InferenceService
+			isvc := defaultIsvc(serviceKey.Namespace, serviceKey.Name, storageUri)
+			isvc.Annotations[constants.StopAnnotationKey] = "true"
+			Expect(k8sClient.Create(context.TODO(), isvc)).NotTo(HaveOccurred())
+			defer k8sClient.Delete(ctx, isvc)
+
+			// Check that the KSVC does not exist
+			actualService := &knservingv1.Service{}
+			predictorServiceKey := types.NamespacedName{
+				Name:      constants.PredictorServiceName(serviceKey.Name),
+				Namespace: serviceKey.Namespace,
+			}
+			Consistently(func() bool {
+				err := k8sClient.Get(context.TODO(), predictorServiceKey, actualService)
+				return apierr.IsNotFound(err)
+			}, timeout).Should(BeTrue(), "The ksvc should not be created")
+
+			// Check that the virtual service is deleted
+			virtualService := &istioclientv1beta1.VirtualService{}
+			Eventually(func() bool {
```

Comment:
Consistently would be the right one to check if the VS will not be created or not.